Automattic/wp-calypso
# Hosting Command Palette: Add command to navigate to WooCommerce

Related to https://github.com/Automattic/dotcom-forge/issues/4859

## Proposed Changes

This PR adds a command to `Open WooCommerce settings`:

* If a site does not have a WooCommerce plugin activated (e.g. sites with plans below `Creator` plan or a site with a `Creator` plan but no store set up), the customer would be sent to `/woocommerce-installation/{yourAtomicSite}`;
* If a site has a WooCommerce plugin activated, the customer will be sent to WooCommerce in `wp-admin`;
* P2 sites and Jetpack sites are filtered out at the moment. @sejas should we include Jetpack sites and redirect to `wp-admin`?

This is to remain consistent with the behavior of the WooCommerce menu item at the moment.

<img width="1074" alt="Screenshot 2023-12-20 at 6 02 14 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/67796c7b-03df-4551-aac1-47846d3f5785">

## Testing Instructions

**To test an Atomic site with a store set up**: 

* Navigate to Calypso Live Link
* Press `cmd + k` on macOS or `ctrl + k` on Windows to open the command palette
* Search for the `Open WooCommerce settings` command
* Select an Atomic site with the store set up 
* Confirm that you are brought to `wp-admin/admin.php?page=wc-admin`

**To test with on a site with no store**: 

* Navigate to Calypso Live Link
* Press `cmd + k` on macOS or `ctrl + k` on Windows to open the command palette
* Search for the `Open WooCommerce settings` command
* Select a site with an Explorer plan or a Free plan
* Confirm that you are brought to `/woocommerce-installation/${ site.slug }`

## Pre-merge Checklist
